Okwme/80 target

okwme opened the pull request:
closes #80 and #79 

Options added:
```js
new Anybody(p, {
  target: "outside", // "outside" or "inside", when inside it's not target pattern
  showLives: true, // true or false, when true you get a wreath of circles around each body showing levels
})
```

The four variations look as follows:
![Screenshot 2024-04-02 at 17 37 01](https://github.com/okwme/anybody-problem/assets/964052/b7094d6a-fd60-4fcd-92d6-636b8716d6f9)
![Screenshot 2024-04-02 at 17 12 18](https://github.com/okwme/anybody-problem/assets/964052/e5729e41-2132-4486-9c07-7285b424b643)
![Screenshot 2024-04-02 at 17 11 28](https://github.com/okwme/anybody-problem/assets/964052/484a37b6-c44e-4a09-b5ef-2f04f4a2cc77)
![Screenshot 2024-04-02 at 17 07 19](https://github.com/okwme/anybody-problem/assets/964052/7b7764d3-ea53-48c4-96a0-9b96537a302d)


I prefer `target: "outside"` and `showLives: true`. What do yall think @psugihara @evvvritt ?

Also added cross-hair cursor and drew the gun all the way out to the cursor. Animated the gun too, this actually kind of helps in aiming, maybe we sell this as an add-on feature?

Currently rotating lives are just circles but Joon suggested we use the small star symbol that will eventually take their place in the background which I like.

psugihara commented on src/visuals.js:
```diff
@@ -383,26 +405,45 @@ export const Visuals = {
+    const drawingContext = this.p.canvas.getContext('2d')
+    const chunk = this.windowWidth / 100
+    drawingContext.setLineDash([chunk])
+    drawingContext.lineDashOffset = -(this.frames * 10)
```
What's the animation that's supposed to happen here with the dashed line? What I see is it shakes then stops then shakes then stops but timing doesn't seem consistent.

I like the line idea but find the animation a little confusing. Maybe kill it or make it slower by removing the `* 10`?

okwme replied:
ah interesting, you use safari right? maybe looks different there. p5 doesn't have any dashed line capability but native canvas context has it so I add line dash of chunk size, then I offset it based on the frame number so that it's a moving dashed line. I set it to 10 because then it matches the speed of the missile as it leaves the gun. Actually helped match up speed and direction on my machine but that's when it was working correctly. I'll take a look on Safari. Wanted to disable by default anyway tho and make it possibly like a paid option. Can add that here already.

okwme commented:
>the tail being kind of continuous from the body make the circle circles feel a little off-center. seems fine though, i don't have a better idea

the target circle feels off center? Should be balanced right at the top of the head, although quick turns spin the head around the target really quickly.

> i like the outside target but it might be nice to put it in one of the small circles so that stuff connects better and you'd have one less thing moving independently (calm++)

As in put the target circle into one of the rotating circles? I would love to do this and actually had discussed it as an option but it completely changes the physics part, since it's actually the target that contains "mass" and applies gravity on the other targets. By making it spin around we'd lose the abiliity to "shoot" it at the circuit level : \  One other option was to change the circuit so that the target was the size of the body but I think that would make shooting them too easy.

> if you make that change, the white filled one could more directly connect to the target it replaces... maybe make it a faded target or target with a checkmark or something. It took me a few refreshes to understand that means i hit it already.

I tried a few styles for the filled in target but nothing was working great (current solution included). One other one I tried was colored in but semi-trasnparent for past levels and just won level was colored but fully opaque. Couldn't really tell unfortunately. Think we should open up another issue for variations on level style but if the ring is close enough in the direction I'd keep current styles for now.

okwme commented:
added option for aim helper and added new issue to discuss options for level visualisations here:  
https://github.com/okwme/anybody-problem/issues/87

I see what you mean on safari the gun looks much more like it's flashing but actually i think it's just got a worse refresh frame rate thing because when you shoot it still matches the pace. Maybe it's not helpful anyway but the default option is to have it turned off now.

psugihara commented:
> As in put the target circle into one of the rotating circles? I would love to do this and actually had discussed it as an option but it completely changes the physics part, since it's actually the target that contains "mass" and applies gravity on the other targets. By making it spin around we'd lose the abiliity to "shoot" it at the circuit level : \ One other option was to change the circuit so that the target was the size of the body but I think that would make shooting them too easy.

Ah OK I wasn't thinking about circuit constraints, just spitballing how to make the visual more coherent.
